src/utils/manager.py

```diff
@@ -237,3 +293,56 @@ def umount(mountpoint: Union[str, os.PathLike], force: bool = False) -> None:
     except subprocess.TimeoutExpired as e:
         _logger.error(f"{e} Reason:\n{e.stderr}")
         raise Error(f"Failed to unmount {mountpoint} failed after {e.timeout:.0f}s")
+
+
+def _trigger_autofs() -> None:
+    """Triggers a mount on all filesystems handled by autofs.
+
+    Raises:
+        Error: Raised if `findmnt` is unable to return info about the automounted filesystems.
+    """
+    try:
+        proc = subprocess.run(
+            ["findmnt", "-J", "-t", "autofs", "-o", "TARGET"],
+            capture_output=True,
+            check=True,
+            text=True,
+            encoding="utf-8",
+        )
+
+        # Old versions of `findmnt` only output an empty string if
+        # the list of mounts is empty, which is definitely not valid JSON.
+        output = proc.stdout.strip() or '{ "filesystems": [] }'
+
+        filesystems = json.loads(output)["filesystems"]
+        for fs in filesystems:
+            target = fs["target"]
+            _logger.info(f"triggering automount for `{target}`")
+
+            try:
+                # TODO: Check if there's a better way to trigger an automount.
+                with os.scandir(target) as it:
+                    next(it)
+            except OSError as e:
+                _logger.error(f"Could not trigger automount for `{target}`. Reason:\n{e}")
+
+    except subprocess.CalledProcessError as e:
+        _logger.error(f"{e} Reason:\n{e.stderr}")
+        raise Error("Failed to call command `findfs`")
+    except json.JSONDecodeError as e:
+        _logger.error(f"Failed to parse output of `findmnt`. Reason:\n{e}")
+        raise Error("Failed to parse output of `findmnt`")
+
+
+def _mountpoint_to_autofs_id(mountpoint: Union[str, os.PathLike]) -> str:
+    """Converts a mountpoint to its corresponding autofs id.
+
+    Args:
+        mountpoint: NFS share mountpoint.
+    """
+    # Cannot use `removeprefix` since we need to support python <3.8 for Ubuntu 20.04
+    path = pathlib.Path(mountpoint).resolve().as_posix()
+    if path.startswith("/"):
+        path = path[1:]
+
+    return path.replace("/", "-")
```

Comment:
Nice idea creating a unique id for each of the shares that we are mounting. Prevents any potential cross conflicts if we have multiple NFS shares mounted on the same device. A couple notes:

1. Why are you using the `as_posix()` method here? The charm is exclusively deployed to systems that use forward slashes as the path separator. If you're using `as_posix()` to exclusively convert the `PosixPath` object to a string, you can just directly cast the object to a string by using the `str(...)` built-in.
2. Why not use `str.lstrip("/")`? That way you don't need any nesting. You're implementation could be the following:

```suggestion
    path = pathlib.Path(mountpoint).resolve()
    path = str(path).lstrip("/")
    return path.replace("/", "-")
```

Reply:
I originally used the `as_posix` to make it obvious that we wanted forward slashes, but you're right that this will only be used on *nix systems, so it's just better to use `str`.